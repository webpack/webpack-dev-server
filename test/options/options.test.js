'use strict';

const options = require('../../lib/options.json');

describe('options', () => {
  it('should return properties', () => {
    expect(options.properties).toMatchSnapshot();
  });

  it('should return errorMessage', () => {
    expect(options.errorMessage.properties).toMatchSnapshot();
  });

  it('should match properties and errorMessage', () => {
    const properties = Object.keys(options.properties);
    const messages = Object.keys(options.errorMessage.properties);

    expect(properties.length).toEqual(messages.length);

    const res = properties.every((name) => messages.includes(name));

    expect(res).toEqual(true);
  });
});
