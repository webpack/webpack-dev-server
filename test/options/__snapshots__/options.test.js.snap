// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options should return errorMessage 1`] = `
Object {
  "after": "should be {Function} (https://webpack.js.org/configuration/dev-server/#devserverafter)",
  "allowedHosts": "should be {Array} (https://webpack.js.org/configuration/dev-server/#devserverallowedhosts)",
  "before": "should be {Function} (https://webpack.js.org/configuration/dev-server/#devserverbefore)",
  "bonjour": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverbonjour)",
  "ca": "should be {String|Buffer}",
  "cert": "should be {String|Buffer}",
  "clientLogLevel": "should be {String} and equal to one of the allowed values

 [ 'info', 'warn', 'error', 'debug', 'trace', 'silent' ]

 (https://webpack.js.org/configuration/dev-server/#devserverclientloglevel)",
  "compress": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devservercompress)",
  "contentBase": "should be {Number|String|Array} (https://webpack.js.org/configuration/dev-server/#devservercontentbase)",
  "disableHostCheck": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverdisablehostcheck)",
  "features": "should be {Array}",
  "filename": "should be {String|RegExp|Function} (https://webpack.js.org/configuration/dev-server/#devserverfilename-)",
  "fs": "should be {Object} (https://github.com/webpack/webpack-dev-middleware#fs)",
  "headers": "should be {Object} (https://webpack.js.org/configuration/dev-server/#devserverheaders-)",
  "historyApiFallback": "should be {Boolean|Object} (https://webpack.js.org/configuration/dev-server/#devserverhistoryapifallback)",
  "host": "should be {String|Null} (https://webpack.js.org/configuration/dev-server/#devserverhost)",
  "hot": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverhot)",
  "hotOnly": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverhotonly)",
  "http2": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverhttp2)",
  "https": "should be {Object|Boolean} (https://webpack.js.org/configuration/dev-server/#devserverhttps)",
  "index": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserverindex)",
  "injectClient": "should be {Boolean|Function}",
  "injectHot": "should be {Boolean|Function}",
  "inline": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverinline)",
  "key": "should be {String|Buffer}",
  "lazy": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverlazy-)",
  "log": "should be {Function}",
  "logLevel": "should be {String} and equal to one of the allowed values

 [ 'info', 'warn', 'error', 'debug', 'trace', 'silent' ]

 (https://github.com/webpack/webpack-dev-middleware#loglevel)",
  "logTime": "should be {Boolean} (https://github.com/webpack/webpack-dev-middleware#logtime)",
  "mimeTypes": "should be {Object} (https://webpack.js.org/configuration/dev-server/#devservermimetypes-)",
  "noInfo": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devservernoinfo-)",
  "open": "should be {String|Boolean} (https://webpack.js.org/configuration/dev-server/#devserveropen)",
  "openPage": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserveropenpage)",
  "overlay": "should be {Boolean|Object} (https://webpack.js.org/configuration/dev-server/#devserveroverlay)",
  "pfx": "should be {String|Buffer} (https://webpack.js.org/configuration/dev-server/#devserverpfx)",
  "pfxPassphrase": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserverpfxpassphrase)",
  "port": "should be {Number|String|Null} (https://webpack.js.org/configuration/dev-server/#devserverport)",
  "progress": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverprogress---cli-only)",
  "proxy": "should be {Object|Array} (https://webpack.js.org/configuration/dev-server/#devserverproxy)",
  "public": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserverpublic)",
  "publicPath": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserverpublicpath-)",
  "quiet": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverquiet-)",
  "reporter": "should be {Function} (https://github.com/webpack/webpack-dev-middleware#reporter)",
  "requestCert": "should be {Boolean}",
  "serveIndex": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverserveindex)",
  "serverSideRender": "should be {Boolean} (https://github.com/webpack/webpack-dev-middleware#serversiderender)",
  "setup": "should be {Function} (https://webpack.js.org/configuration/dev-server/#devserversetup)",
  "sockPath": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserversockpath)",
  "sockPort": "should be {Number|String|Null}",
  "socket": "should be {String} (https://webpack.js.org/configuration/dev-server/#devserversocket)",
  "staticOptions": "should be {Object} (https://webpack.js.org/configuration/dev-server/#devserverstaticoptions)",
  "stats": "should be {Object|Boolean} (https://webpack.js.org/configuration/dev-server/#devserverstats-)",
  "useLocalIp": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserveruselocalip)",
  "warn": "should be {Function}",
  "watchContentBase": "should be {Boolean} (https://webpack.js.org/configuration/dev-server/#devserverwatchcontentbase)",
  "watchOptions": "should be {Object} (https://webpack.js.org/configuration/dev-server/#devserverwatchoptions-)",
  "writeToDisk": "should be {Boolean|Function} (https://webpack.js.org/configuration/dev-server/#devserverwritetodisk-)",
}
`;

exports[`options should return properties 1`] = `
Object {
  "after": Object {
    "instanceof": "Function",
  },
  "allowedHosts": Object {
    "items": Object {
      "type": "string",
    },
    "type": "array",
  },
  "before": Object {
    "instanceof": "Function",
  },
  "bonjour": Object {
    "type": "boolean",
  },
  "ca": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "instanceof": "Buffer",
      },
    ],
  },
  "cert": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "instanceof": "Buffer",
      },
    ],
  },
  "clientLogLevel": Object {
    "enum": Array [
      "info",
      "warn",
      "error",
      "debug",
      "trace",
      "silent",
    ],
  },
  "compress": Object {
    "type": "boolean",
  },
  "contentBase": Object {
    "anyOf": Array [
      Object {
        "enum": Array [
          false,
        ],
      },
      Object {
        "type": "number",
      },
      Object {
        "type": "string",
      },
      Object {
        "items": Object {
          "type": "string",
        },
        "minItems": 1,
        "type": "array",
      },
    ],
  },
  "disableHostCheck": Object {
    "type": "boolean",
  },
  "features": Object {
    "items": Object {
      "type": "string",
    },
    "type": "array",
  },
  "filename": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "instanceof": "RegExp",
      },
      Object {
        "instanceof": "Function",
      },
    ],
  },
  "fs": Object {
    "type": "object",
  },
  "headers": Object {
    "type": "object",
  },
  "historyApiFallback": Object {
    "anyOf": Array [
      Object {
        "type": "boolean",
      },
      Object {
        "type": "object",
      },
    ],
  },
  "host": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "type": "null",
      },
    ],
  },
  "hot": Object {
    "type": "boolean",
  },
  "hotOnly": Object {
    "type": "boolean",
  },
  "http2": Object {
    "type": "boolean",
  },
  "https": Object {
    "anyOf": Array [
      Object {
        "type": "object",
      },
      Object {
        "type": "boolean",
      },
    ],
  },
  "index": Object {
    "type": "string",
  },
  "injectClient": Object {
    "anyOf": Array [
      Object {
        "type": "boolean",
      },
      Object {
        "instanceof": "Function",
      },
    ],
  },
  "injectHot": Object {
    "anyOf": Array [
      Object {
        "type": "boolean",
      },
      Object {
        "instanceof": "Function",
      },
    ],
  },
  "inline": Object {
    "type": "boolean",
  },
  "key": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "instanceof": "Buffer",
      },
    ],
  },
  "lazy": Object {
    "type": "boolean",
  },
  "log": Object {
    "instanceof": "Function",
  },
  "logLevel": Object {
    "enum": Array [
      "info",
      "warn",
      "error",
      "debug",
      "trace",
      "silent",
    ],
  },
  "logTime": Object {
    "type": "boolean",
  },
  "mimeTypes": Object {
    "type": "object",
  },
  "noInfo": Object {
    "type": "boolean",
  },
  "open": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "type": "boolean",
      },
    ],
  },
  "openPage": Object {
    "type": "string",
  },
  "overlay": Object {
    "anyOf": Array [
      Object {
        "type": "boolean",
      },
      Object {
        "properties": Object {
          "errors": Object {
            "type": "boolean",
          },
          "warnings": Object {
            "type": "boolean",
          },
        },
        "type": "object",
      },
    ],
  },
  "pfx": Object {
    "anyOf": Array [
      Object {
        "type": "string",
      },
      Object {
        "instanceof": "Buffer",
      },
    ],
  },
  "pfxPassphrase": Object {
    "type": "string",
  },
  "port": Object {
    "anyOf": Array [
      Object {
        "type": "number",
      },
      Object {
        "type": "string",
      },
      Object {
        "type": "null",
      },
    ],
  },
  "progress": Object {
    "type": "boolean",
  },
  "proxy": Object {
    "anyOf": Array [
      Object {
        "type": "object",
      },
      Object {
        "items": Object {
          "anyOf": Array [
            Object {
              "type": "object",
            },
            Object {
              "instanceof": "Function",
            },
          ],
        },
        "minItems": 1,
        "type": "array",
      },
    ],
  },
  "public": Object {
    "type": "string",
  },
  "publicPath": Object {
    "type": "string",
  },
  "quiet": Object {
    "type": "boolean",
  },
  "reporter": Object {
    "instanceof": "Function",
  },
  "requestCert": Object {
    "type": "boolean",
  },
  "serveIndex": Object {
    "type": "boolean",
  },
  "serverSideRender": Object {
    "type": "boolean",
  },
  "setup": Object {
    "instanceof": "Function",
  },
  "sockPath": Object {
    "type": "string",
  },
  "sockPort": Object {
    "anyOf": Array [
      Object {
        "type": "number",
      },
      Object {
        "type": "string",
      },
      Object {
        "type": "null",
      },
    ],
  },
  "socket": Object {
    "type": "string",
  },
  "staticOptions": Object {
    "type": "object",
  },
  "stats": Object {
    "anyOf": Array [
      Object {
        "type": "object",
      },
      Object {
        "type": "boolean",
      },
      Object {
        "enum": Array [
          "none",
          "errors-only",
          "minimal",
          "normal",
          "verbose",
        ],
      },
    ],
  },
  "useLocalIp": Object {
    "type": "boolean",
  },
  "warn": Object {
    "instanceof": "Function",
  },
  "watchContentBase": Object {
    "type": "boolean",
  },
  "watchOptions": Object {
    "type": "object",
  },
  "writeToDisk": Object {
    "anyOf": Array [
      Object {
        "type": "boolean",
      },
      Object {
        "instanceof": "Function",
      },
    ],
  },
}
`;
